Character.cpp

```diff
@@ -0,0 +1,37 @@
+#include "Character.h"
+#include <iostream>
+
+//constructor
+Character::Character(std::string char_name,  int char_hp, int char_dmg ) : name(char_name),hp(char_hp), damage(char_dmg) {
+}
+//getters
+std::string Character::getName() const {
+	return name;
+}
+int Character::getHp() const {
+	return hp;
+}
+int Character::getDamage() const{
+	return damage;
+}
+//func
+void Character::attack(Character&  target) const{
+	if (target.getHp() < this->getDamage()){
+		target.hp=0;
+	}
+	else{
+		target.hp=target.getHp()-this->getDamage();	
+	}
```

Comment:
Hm.. kb fel oraja lattam ugyanennel a fuggvenynel ugyanezt a fura { } konvenciot. es ott is tok foloslegesen ott volt a this->. Ezt miert igy csinaljatok? Nem rossz, csak minel tobb a kod, annal nehezebb atlatni sokszor. Illetve a -= hasznos dolog.

Reply:
Ha más folytatja a kódot, talán egyszerűbben átlátható, de átírtam.